drivers/scheduler/k8s/k8s.go

```diff
@@ -7517,3 +7528,25 @@ func ClusterVersion() (string, error) {
 	}
 	return strings.TrimLeft(ver.String(), "v"), nil
 }
+
+func createClonedStorageClassIfRequired(originalStorageClass *storageapi.StorageClass) (string, error) {
+	clonedSCName := originalStorageClass.Name + "-clone"
+	clonedSCobj, err := k8sStorage.GetStorageClass(clonedSCName)
+	if err == nil {
+		return clonedSCobj.Name, nil
+	} else {
+		log.Infof("Cloned SC with the name %v does not exist and has to be created", clonedSCName)
+		originalStorageClass.Name = clonedSCName
+		immediate := storageapi.VolumeBindingImmediate
+		originalStorageClass.VolumeBindingMode = &immediate
+		originalStorageClass.ResourceVersion = ""
+		originalStorageClass.UID = ""
+
+		clonedSCobj, err := k8sStorage.CreateStorageClass(originalStorageClass)
+		if err != nil {
+			log.Infof("Error occurred while creating cloned storage class:- %v", err)
+			return "", fmt.Errorf("failed to create cloned storage class with the name %s : %s", originalStorageClass.Name, err)
```

Comment:
nit: "failed to create cloned storage class" should be added by the caller. 
It would be more clear to consistently use error reporting for errors that should be handled by the caller, and use logging for informational messages or errors that this function can recover from.

Reply:
fixed.